api_v3/lib/KalturaLiveEntryService.php

```diff
@@ -246,26 +246,40 @@ function registerMediaServerAction($entryId, $hostname, $mediaServerIndex, $appl
 		}
 
 		$dbEntry->setRedirectEntryId(null);
-		
 		if($dbEntry->save())
 		{
 			if($mediaServerIndex == MediaServerIndex::PRIMARY && $dbEntry->getRecordStatus())
 			{
+				KalturaLog::info("Checking if recorded entry needs to be created for entry $entryId");
 				$createRecordedEntry = false;
 				if(!$dbEntry->getRecordedEntryId())
 				{
 					$createRecordedEntry = true;
+					KalturaLog::info("Creating new recoreded entry for $entryId ");
 				}
-				elseif($dbEntry->getRecordStatus() == RecordStatus::PER_SESSION && ($dbEntry->getLastBroadcastEndTime() + kConf::get('live_session_reconnect_timeout', 'local', 180) < time()))
-				{
-					$createRecordedEntry = true;
+				else {
+					$dbRecordedEntry = entryPeer::retrieveByPK($dbEntry->getRecordedEntryId());
+					if (!$dbRecordedEntry) {
+						throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $dbRecordedEntry);
```

Comment:
Since "$dbEntry->getRecordedEntryId()" exists (i.e. not null), the assumption is that the entry exists.
However, since the API server is a distributed and complicated system, we want to guarantee  that  the API call would not return null.
The idea is similar to the code written in line  230 ( i.e. "if (!$dbEntry || !($dbEntry instanceof LiveEntry)" ), and in consultation with Elad Cohen.